cypress/support/step-definitions/carousel-steps.js

```diff
@@ -23,9 +23,17 @@ function clickCarouselButton(direction) {
   }
 }
 
-Then('slide title is {string}', (title) => {
+Then('classic slide title is {string}', (title) => {
   cy.wait(1500, { log: DEBUG_FLAG }); // required because of component refresh
-  slide().should('have.text', title);
+  classicSlide().should('have.text', title);
+});
+
+Then('slide {int} title is {string}', (index, title) => {
+  cy.wait(1500, { log: DEBUG_FLAG }); // required because of component refresh
+  for (let i = 1; i < index; ++i) {
```

Comment:
instead of `for` loop use `slide(index).should('have.text', title);`  
First change locator which are proposed :)

Reply:
Amended using your solution 👍